metrics/cli/test/cli/updater/updater_test.dart

```diff
@@ -0,0 +1,305 @@
+    final directory = DirectoryMock();
+    final updater = Updater(
+      updateAlgorithm: updateAlgorithm,
+      fileHelper: fileHelper,
+      pathsFactory: pathsFactory,
+      prompter: prompter,
+    );
+    final config = _UpdateConfigMock();
+
+    PostExpectation<Directory> whenCreateTempDirectory() {
+      return when(fileHelper.createTempDirectory(any, any));
+    }
+
+    PostExpectation<bool> whenDirectoryExist({
+      Directory withDirectory,
+      String withPath,
+    }) {
+      final currentDirectory = withDirectory ?? directory;
+      whenCreateTempDirectory().thenReturn(currentDirectory);
+
+      final currentPath = withPath ?? tempDirectoryPath;
+      when(currentDirectory.path).thenReturn(currentPath);
+
+      return when(currentDirectory.existsSync());
+    }
+
+    tearDown(() {
+      reset(updateAlgorithm);
+      reset(fileHelper);
+      reset(prompter);
+      reset(pathsFactoryMock);
+      reset(directory);
+    });
+
+    test(
+      "throws an ArgumentError if the given update algorithm is null",
+      () {
+        expect(
+          () => Updater(
+            updateAlgorithm: null,
+            fileHelper: fileHelper,
+            prompter: prompter,
+            pathsFactory: pathsFactory,
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given FileHelper is null",
+      () {
+        expect(
+          () => Updater(
+            updateAlgorithm: updateAlgorithm,
+            fileHelper: null,
+            prompter: prompter,
+            pathsFactory: pathsFactory,
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given Prompter is null",
+      () {
+        expect(
+          () => Updater(
+            updateAlgorithm: updateAlgorithm,
+            fileHelper: fileHelper,
+            prompter: null,
+            pathsFactory: pathsFactory,
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "throws an ArgumentError if the given PathsFactory is null",
+      () {
+        expect(
+          () => Updater(
+            updateAlgorithm: updateAlgorithm,
+            fileHelper: fileHelper,
+            prompter: prompter,
+            pathsFactory: null,
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".update() creates a temporary directory",
+      () async {
+        whenDirectoryExist().thenReturn(true);
+
+        await updater.update(config);
+
+        verify(fileHelper.createTempDirectory(any, prefix)).called(once);
+      },
+    );
+
+    test(
+      ".update() creates a temporary directory before creating the Paths instance",
+      () async {
+        final updater = Updater(
+          updateAlgorithm: updateAlgorithm,
+          fileHelper: fileHelper,
+          pathsFactory: pathsFactoryMock,
+          prompter: prompter,
+        );
```

Comment:
✅ 
Should we introduce the default `updater` instance within the group of test cases so we shouldn't recreate it each time?